### ISSUE TITLE
ELI5 on color theming with SASS?

### ISSUE THREAD

**Issue by deronsizemore** [twbs/bootstrap#27644]
I'm just getting started with Bootstrap 4 and SASS so please bear with me.

I have my .scss file which I'm creating my styles in. I'm not quite sure I understand the best approach to theme my site.

I started out doing something like this in my .scss file:

```
// colors
$blue: #3498db;
$green: #37bf91;
$red: #f75676;
$orange: #febd12;
$white: #ffffff;
$gray: #7C8D98;
$gray-dark: #4A545A;

.my-class {
    color: $blue;
}
```

That's how I've been theming this design so far. My confusion is from seeing references to color maps in the documentation and online. I'm not quite sure if/when/why I'd use the below in my own .scss file? 

```
$colors: (
  "blue": $blue,
  "indigo": $indigo,
  "purple": $purple,
  "pink": $pink,
  "red": $red,
  "orange": $orange,
  "yellow": $yellow,
  "green": $green,
  "teal": $teal,
  "cyan": $cyan,
  "white": $white,
  "gray": $gray-600,
  "gray-dark": $gray-800
) !default;
```

or 

```
$theme-colors: (
    "primary":    $primary,
    "secondary":  $secondary,
    "success":    $success,
    "info":       $info,
    "warning":    $warning,
    "danger":     $danger,
    "light":      $light,
    "dark":       $dark
) !default;
```

I guess I'm just looking for a "like I'm five" on how to best approach customizing colors in my own .scss file. Thank you.

**Comment by wolfy1339** [comment]
The color maps aren't really used at the moment, the goal was to provide a color palette with it.
You can just use the direct color variables just fine, as the color map will get updated from the variables

**Comment by deronsizemore** [comment]
@wolfy1339 Thank you. So in the end, I can just set my own variables for colors and really name them whatever I want? They don't necessarily need to be named "primary," "secondary," "blue," etc.? I could do a variable like: `nav-bar-color: #444;` and it would work all the same?

**Comment by AceJJ** [comment]
You can name SASS variables whatever you want in your own custom scss. The way I would do it, is to use 'nav-bar-color' as the color for the nav-bar, but then for a color for some other element that matches the nav bar, I'd make a new variable, like:

`$button-primary-color : $nav-bar-color;`

This way, if I ever decided I wanted to change the primary color of buttons, I could just change the value of button primary. If I used nav-bar-color everywhere, it would be much harder to change the color of elements that match nav bar.

**Comment by deronsizemore** [comment]
@AceJJ Ah, thank you. That makes perfect sense.

**Comment by deronsizemore** [comment]
Had one other question as a result of looking through my compiled .css file. 

I have this inside that file:

```
:root {
  --blue: #3498db;
  --indigo: #6610f2;
  --purple: #6f42c1;
  --pink: #e83e8c;
  --red: #f75676;
  --orange: #febd12;
  --yellow: #fff4b1;
  --green: #37bf91;
  --teal: #20c997;
  --cyan: #17a2b8;
  --white: #ffffff;
  --gray: #6c757d;
  --gray-dark: #343a40;
  --primary: #3498db;
  --secondary: #6c757d;
  --success: #37bf91;
  --info: #17a2b8;
  --warning: #fff4b1;
  --danger: #f75676;
  --light: #eef2f5;
  --dark: #4a545a;
```

The color variables from my SCSS file are:

```
// colors
$blue: #3498db;
$green: #37bf91;
$red: #f75676;
$orange: #febd12;
$yellow: #fff4b1;
$white: #ffffff;
$gray: #DDE4E8;
$gray-dark: #8597A3;
$light: #eef2f5;
$dark: #4a545a;
```

As you can see, the variables I've set inside my main.scss file are compiled into the main.css file but it looks like the $gray and $gray-dark are not correctly being set in the .css file? Why would those two not update correctly? My live site seems to be fine and seems to be using the variables as I've set them inside my .scss file so it was confusing why they aren't actually showing correctly in that color list at the top of the .css file.
